### PR TITLE
fixed typo mistakes in the new-module file

### PR DESCRIPTION
**Pull request by ibilalkayy**
### What's being changed:

In this Pull Request, I fixed some typo mistakes that were present in the new-module file. Some of them, I added and some of them, I removed.

### Type of change:

- [x] **Documentation** updates (non-breaking change to fix/update documentation)
- [ ] **Website** updates (non-breaking change to update main page, company pages, pricing, etc)
- [ ] **Content** updates – **blog**, **podcast** (non-breaking change to add/update content)
- [ ] **Bug fix** (non-breaking change to fixes an issue with the site)
- [ ] **Feature** or **enhancements** (non-breaking change to add functionality)

### How Has This Been Tested?

- [ ] **Github action** – automated build completed without errors
- [x] **Local build** - the site works as expected when running `yarn start`

> note, you can run `yarn verify-links` to test site links locally

### CODE REVIEW

File: developers/contributor-guide/weaviate-modules/how-to-build-a-new-module.md [hunk 1]
```diff
@@ -38,16 +38,16 @@ If you want to make a pull request to Weaviate with your custom module, make sur
...
 
 ## Design the inference model (part 2)
 
 The inference model is a service that provides at least four API endpoints:
-1. `GET /.well-known/live` -> which should respond `204` when the app is alive
-2. `GET /.well-known/ready` -> which should respond `204` when the app is ready to serve traffic
-3. `GET /meta` -> which should respond meta information about the inference model
-4. `POST /<foo>` (at least 1) -> which is the endpoint that the Weaviate Module uses for inference. For a vectorizer this might for example be `POST /vectors`, which takes a JSON body with the data to vectorize. A vector will be returned (in JSON format). The Question Answering model, on the other hand, has an endpoint `POST /answers`, which takes a JSON body with the text to tokenize and returns a list of tokens found in the text (also formatted as JSON).
+1. `GET /.well-known/live` -> that should respond `204` when the app is alive.
```
Comment (daveatweaviate) [inline developers/contributor-guide/weaviate-modules/how-to-build-a-new-module.md:47]: Consider formatting lines 47-49 as

`GET /.well-known/live` responds ...

File: developers/contributor-guide/weaviate-modules/how-to-build-a-new-module.md [hunk 2]
```diff
@@ -38,16 +38,16 @@ If you want to make a pull request to Weaviate with your custom module, make sur
...
+2. `GET /.well-known/ready` -> that should respond `204` when the app is ready to serve traffic.
+3. `GET /meta` -> that should respond meta information about the inference model.
+4. `POST /<foo>` (at least 1) -> that is the endpoint that the Weaviate Module uses for inference. For a vectorizer this might for example be `POST /vectors`, which takes a JSON body with the data to vectorize. A vector will be returned (in JSON format). The Question Answering model, on the other hand, has an endpoint `POST /answers`, which takes a JSON body with the text to tokenize and returns a list of tokens found in the text (also formatted as JSON).
```
Comment (daveatweaviate) [inline developers/contributor-guide/weaviate-modules/how-to-build-a-new-module.md:50]: Consider:
`POST /<foo>` (at least 1) is the endpoint ...

File: developers/contributor-guide/weaviate-modules/how-to-build-a-new-module.md [hunk 3]
```diff
@@ -84,7 +84,7 @@ Make sure to also write tests for the GraphQL field and for the result (e.g. [th
...
+If you choose to add a filter outside the `_additional` GraphQL field, you need to take a slight different approach of adding the filter arguments as explained in the previous step. That is because you can't include the filter arguments in the `/additional` GraphQL field. For example, the QnA module has the filter `ask` on class level (click [here](/developers/weaviate/modules/reader-generator-modules/qna-transformers.md#graphql-ask-search) for an example). This argument was created in a new folder inside the new module folder in Weaviate ([example](https://github.com/weaviate/weaviate/tree/master/modules/qna-transformers/ask)). To achieve this, make sure to follow these steps:
```
Comment (daveatweaviate) [inline developers/contributor-guide/weaviate-modules/how-to-build-a-new-module.md:87]: Consider: 

a slightly different approach to add

File: developers/contributor-guide/weaviate-modules/how-to-build-a-new-module.md [hunk 4]
```diff
@@ -124,9 +124,9 @@ Add user-specific configuration to both the Weaviate module and the inference AP
...
-    3. Add the service to `tools/dev/run_dev_server.sh` ([example](https://github.com/weaviate/weaviate/blob/7036332051486b393d83f9ea2ffb0ca1b2269328/tools/dev/run_dev_server.sh#L77-L92)). This is the Docker Compose setup for running the development server. You should define the modules that you want to run with your new dev setup: where they are running and their configuration. It also includes the command to run Weaviate, you can copy this.
+    3. Add the service to `tools/dev/run_dev_server.sh` ([example](https://github.com/weaviate/weaviate/blob/7036332051486b393d83f9ea2ffb0ca1b2269328/tools/dev/run_dev_server.sh#L77-L92)). This is the Docker Compose setup for running the development server. You should define the modules that you want to run with your new dev setup: where they are running their configuration. It also includes the command to run Weaviate, you can copy this.
```
Comment (daveatweaviate) [inline developers/contributor-guide/weaviate-modules/how-to-build-a-new-module.md:129]: The "and" is needed here. Consider:

Define where the new modules should run and configure them.

File: developers/contributor-guide/weaviate-modules/how-to-build-a-new-module.md [hunk 5]
```diff
@@ -146,4 +146,4 @@ tools/dev/run_dev_server.sh --<LOCAL-CONFIG-NAME>
 You can now load in any sample or test dataset. If you only make changes in the `/modules/<new-module>` folder afterwards, you only need to re-run `tools/dev/run_dev_server.sh --<LOCAL-CONFIG-NAME>` to apply the changes. The data will be kept, so no need to re-import.
 
 #### Passing tests
-Finally, and before you make a PR, your Weaviate with your complete module implementation should pass all tests (all existing tests and tests added by you). How to run tests, [check this page](../weaviate-core/tests.md#run-the-whole-pipeline).
+Finally, and before you make a PR, your Weaviate with your complete module implementation should pass all tests (all existing tests added by you). How to run tests, [check this page](../weaviate-core/tests.md#run-the-whole-pipeline).
```
Comment (daveatweaviate) [inline developers/contributor-guide/weaviate-modules/how-to-build-a-new-module.md:149]: The "and" is needed here. Consider: 

Before you submit your PR, your new module implementation must pass all existing tests and any new tests that you added. 

For details on how to run tests, see [Run the whole pipeline](../weaviate-core/tests.md#run-the-whole-pipeline).

**Review by daveatweaviate — COMMENTED** [review]
@ibilalkayy thank you for your suggestions. I left a couple of small comments.

Would you accept the [Contributor License Agreement](https://weaviate.io/service/contributor-license-agreement) in this PR?

I know you agreed to it in #1504 but we need an acknowledgement for each PR.

Thanks!

**Comment by ibilalkayy** [comment]
I agree with the CLA.

**Comment by daveatweaviate** [comment]
@ibilalkayy thanks for your suggestions. I've merged these in.

**Comment by ibilalkayy** [comment]
Thank you, @daveatweaviate for merging this! Can you take a look at this [issue](https://github.com/weaviate/weaviate-io/issues/1514) which I want to work on?

**Comment by daveatweaviate** [comment]
Bilal -

We're taking a look at the 'Edit this page' updates in the context of
overall site design.

That functionality is turned on global through a site configuration
setting, so if we do turn it back on it will likely be a different PR.

Dave

**Comment by ibilalkayy** [comment]
@daveatweaviate Do you want to work on this functionality in the future because I have sent some PRs relevant to it but if you want to get this done, I would love to work on the remaining pages also and put this functionality in them.

Thank you!

**Comment by daveatweaviate** [comment]
Bilal -

Thanks for your interest!

If we reactive the edit links, it's a one or two line change in one of our
build config files. That config change inserts the edit link on all of the
pages. There won't be updates for individual pages.

We still haven't decided, but it looks like the links may be activated
sometime next week.

Regards,

Dave

**Comment by ibilalkayy** [comment]
@daveatweaviate But if the team wants to, I can make those updates on the individual pages and keep you updated on those pages.

**Comment by daveatweaviate** [comment]
Bilal -

Thanks for the suggestion, but the pages don't need to be updated
individually. The build system inserts the change automatically when it
renders the pages. If we updated the individual pages, the edit link would
appear twice on the rendered pages.

Regards,

Dave

**Pull request merged**